src/spatial_reaction_systems/spatial_ODE_systems.jl

```diff
@@ -0,0 +1,182 @@
+
+    # Creates ODEProblem.
+    ofun = build_odefunction(lrs, vert_ps, edge_ps, jac, sparse)        # Builds the ODEFunction.
+    return ODEProblem(ofun, u0, tspan, vert_ps, args...; kwargs...)     # Creates a normal ODEProblem.
+end
+
+# Builds an ODEFunction for a spatial ODEProblem.
+function build_odefunction(lrs::LatticeReactionSystem, vert_ps::Vector{Vector{T}},
+                           edge_ps::Vector{Vector{T}}, use_jac::Bool, sparse::Bool) where {T}
+    # Prepares (non-spatial) ODE functions and list of spatially moving species and their rates.
+    ofunc = ODEFunction(convert(ODESystem, lrs.rs); jac = use_jac, sparse = false)                  # Creates the (non-spatial) ODEFunction corresponding to the (non-spatial) reaction network.
+    ofunc_sparse = ODEFunction(convert(ODESystem, lrs.rs); jac = use_jac, sparse = true)            # Creates the same function, but sparse. Could insert so this is only computed for sparse cases.
+    transport_rates_speciesmap = compute_all_transport_rates(vert_ps, edge_ps, lrs)                 # Creates a map (Vector{Pair}), mapping each species that is transported to a vector with its transportation rate. If the rate is uniform across all edges, the vector will be length 1 (with this value), else there will be a separate value for each edge.
+    transport_rates = Pair{Int64, Vector{T}}[findfirst(isequal(spat_rates[1]), species(lrs)) => spat_rates[2]
+                       for spat_rates in transport_rates_speciesmap]                                # Remakes "transport_rates_speciesmap". Rates are identical, but the species are represented as their index (in the species(::ReactionSystem) vector). In "transport_rates_speciesmap" they instead were Symbolics. Pair{Int64, Vector{T}}[] is required in case vector is empty (otherwise it becomes Any[], causing type error later).
+
+    f = LatticeDiffusionODEf(ofunc, vert_ps, transport_rates, lrs)                                  # Creates a functor for the ODE f function (incorporating spatial and non-spatial reactions).
+    jac_prototype = (use_jac || sparse) ?
+                    build_jac_prototype(ofunc_sparse.jac_prototype, transport_rates,
+                                        lrs; set_nonzero = use_jac) : nothing                       # Computes the Jacobian prototype (nothing if `jac=false`). 
+    jac = use_jac ? LatticeDiffusionODEjac(ofunc, vert_ps, lrs, jac_prototype, sparse) : nothing    # (Potentially) Creates a functor for the ODE Jacobian function (incorporating spatial and non-spatial reactions).
+    return ODEFunction(f; jac = jac, jac_prototype = (sparse ? jac_prototype : nothing))            # Creates the ODEFunction used in the ODEProblem.
```

Comment:
I'm still a bit confused on the logic here. If the user asks for a jacobian at all you make a sparse prototype even if they didn't request it to be sparse? Then if they don't want a sparse Jacobian but do want a Jacobian you don't give any prototype to `ODEFunction`?

Reply:
I have cleaned this up, should be much clearer what is needed now (basically, if we want a Jacobian, we need to compute both sparse/non-sparse jacobian as they are used as input to various stuff. 

E.g. the `build_jac_prototype` assumes the input (non-spatial) is sparse. I could make two versions of this function (one for sparse input and one for dense), but it seemed easier to just use a sparse jacobian on both cases (and then make the sparse Jac template values dense).

Eitehr case, the rewritten version of the function should be away easier to read.